models/fxdx/optimism/fxdx_optimism_v2_perpetual_trades.sql

```diff
@@ -0,0 +1,127 @@
+                                "fxdx",
+                                \'["kaiblade"]\') }}'
+	)
+}}
+
+
+{% set project_start_date = '2023-05-18' %}
+
+WITH all_executed_positions AS (
+SELECT *,
+       'Open' AS trade_type
+FROM {{ source('fxdxdex_v2_optimism', 'Vault_evt_IncreasePosition') }}
+WHERE evt_tx_hash IN (
+    SELECT evt_tx_hash
+    FROM {{ source('fxdxdex_v2_optimism', 'PositionRouter_evt_ExecuteIncreasePosition') }}
+    )
+    {% if not is_incremental() %}
+  AND evt_block_time >= DATE '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+  AND evt_block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
+    {% endif %}
+
+UNION
+
+SELECT *, 'Close' AS trade_type
+FROM {{ source('fxdxdex_v2_optimism', 'Vault_evt_DecreasePosition') }}
+WHERE evt_tx_hash IN (
+    SELECT evt_tx_hash
+    FROM {{ source('fxdxdex_v2_optimism', 'PositionRouter_evt_ExecuteDecreasePosition') }}
+    )
+    {% if not is_incremental() %}
+  AND evt_block_time >= DATE '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+  AND evt_block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
+    {% endif %}
+),
+
+margin_fees_info AS (
+SELECT *,
+       LEAD(evt_index, 1, 1000000) OVER (PARTITION BY evt_tx_hash ORDER BY evt_index) AS next_evt_index
+FROM {{ source('fxdxdex_v2_optimism', 'Vault_evt_CollectPositionTradeFees') }}
+    {% if not is_incremental() %}
+WHERE evt_block_time >= DATE '{{project_start_date}}'
+    {% endif %}
+    {% if is_incremental() %}
+WHERE evt_block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
+    {% endif %}
+),
+
+complete_perp_tx AS (
+SELECT *, index_token || '/USD' AS market
+FROM (SELECT event.*,
+             (CASE
+                  WHEN collateralToken = 0xd158b0f013230659098e58b66b602dff8f7ff120 THEN 'WETH'
+                  ELSE tokens1.symbol
+                 END)   AS underlying_asset,
+             (CASE
+                  WHEN indexToken = 0xd158b0f013230659098e58b66b602dff8f7ff120 THEN 'ETH'
+                  WHEN tokens.symbol = 'WBTC' THEN 'BTC'
+                  ELSE tokens.symbol
+                 END
+                 )      AS index_token,
+             trx."from",
+             trx.to,
+             fee.feeUsd AS margin_fee
+      FROM all_executed_positions event
+      INNER JOIN {{ source('optimism', 'transactions') }} trx
+          ON event.evt_tx_hash = trx.hash
+          {% if not is_incremental() %}
+          AND evt_block_time >= DATE '{{project_start_date}}'
+          {% endif %}
+          {% if is_incremental() %}
+          AND evt_block_time >= DATE_TRUNC('DAY', NOW() - INTERVAL '7' Day)
+          {% endif %}
+      INNER JOIN margin_fees_info fee
+          ON event.evt_tx_hash = fee.evt_tx_hash
+          AND event.evt_index > fee.evt_index
+          AND event.evt_index < fee.next_evt_index
+      INNER JOIN {{ ref('tokens_optimism_erc20') }} tokens
+          ON event.indexToken = tokens.contract_address
```

Comment:
You can use `tokens_erc20` here with a filter for generic approach.
```suggestion
      INNER JOIN {{ ref('tokens_erc20') }} tokens
          ON event.indexToken = tokens.contract_address
          AND tokens.blockchain = 'optimism'
```